bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -641,19 +642,21 @@ public void onMessageReceived(MiIoSendCommand response) {
                 default:
                     if (refreshListCustomCommands.containsKey(response.getMethod())) {
                         logger.debug("Processing custom refresh command response for !{}", response.getMethod());
-                        MiIoBasicChannel ch = refreshListCustomCommands.get(response.getMethod());
-                        if (response.getResult().isJsonArray()) {
-                            JsonArray cmdResponse = response.getResult().getAsJsonArray();
-                            final String transformation = ch.getTransfortmation();
-                            if (transformation == null || transformation.isBlank()) {
-                                updateChannel(ch, ch.getChannel(),
-                                        cmdResponse.get(0).isJsonPrimitive() ? cmdResponse.get(0)
-                                                : new JsonPrimitive(cmdResponse.get(0).toString()));
+                        final MiIoBasicChannel ch = refreshListCustomCommands.get(response.getMethod());
+                        if (ch != null) {
+                            if (response.getResult().isJsonArray()) {
+                                JsonArray cmdResponse = response.getResult().getAsJsonArray();
+                                final String transformation = ch.getTransfortmation();
+                                if (transformation == null || transformation.isBlank()) {
+                                    updateChannel(ch, ch.getChannel(),
+                                            cmdResponse.get(0).isJsonPrimitive() ? cmdResponse.get(0)
+                                                    : new JsonPrimitive(cmdResponse.get(0).toString()));
```

Comment:
Please cache `cmdResponse.get(0)` in a local variable.

Reply:
agree. done